src/controllers/todo-item.controller.ts

```diff
@@ -0,0 +1,57 @@
+import { repository, EntityCrudRepository } from '@loopback/repository';
+import { TodoItem, TodoItemSchema } from '../models/index';
+import { get, put, param, patch, post, del } from '@loopback/rest';
+
+export class TodoItemController {
+  @repository('todo-item') todoItemRepo: EntityCrudRepository<TodoItem, number>;
+
+  @get('/todo/{id}/items')
+  async getTodoItems(id: number) {
+    return await this.todoItemRepo.find({
+      where: {
+        todoId: id
+      }
+    });
+  }
+
+  @post('/todo/{id}/items')
+  @param.path.number('id')
+  @param.body('todoItem', TodoItemSchema)
+  async createTodoItem(id: number, todoItem: TodoItem) {
+    todoItem.todoId = id;
+    return await this.todoItemRepo.create(todoItem);
+  }
+
+  @put('/todo/{x}/items/{id}')
+  @put('/todoitem/{id}')
+  @param.path.number('id')
+  @param.body('todoItem', TodoItemSchema)
+  @param.path.number('x')
+  async replaceTodoItem(id: number, item: TodoItem, x?: number) {
+    return (await this.todoItemRepo.replaceById(id, item)).toString();
+  }
+
+  @patch('/todo/{x}/items/{id}')
+  @patch('/todoitem/{id}')
+  @param.path.number('id')
+  @param.body('todoItem', TodoItemSchema)
+  @param.path.number('x')
+  async updateTodoItem(id: number, todoItem: TodoItem, x?: number) {
+    // We don't need the todoId, but it's for consistency.
+    return (await this.todoItemRepo.updateById(id, todoItem)).toString();
+  }
+
+  @del('/todo/{x}/items/{id}')
+  @param.path.number('id')
+  async deleteTodoItem(id: number) {
+    return (await this.todoItemRepo.deleteById(id)).toString();
```

Comment:
Is there a good reason for coercing the responses to strings here?

Reply:
The current sequence engine throws when given anything other than a string or object.
Needs a bug fix.

Reply:
Opened issue to track this: https://github.com/strongloop/loopback-next/issues/667